### ISSUE TITLE
Bug in DataDefinition with continuation lines

### ISSUE THREAD

**Issue by fm-117** [TypeCobolTeam/TypeCobol#1671]
**Describe the bug**
Some strange parsing errors may occur when using continuation lines in PICTURE clause.
Here are 2 cases :

**To Reproduce**
(Type)Cobol code that cause the bug : (if any)
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. DVZZMFT0.
      
       DATA DIVISION.
       WORKING-STORAGE SECTION.
      
       01 STR.
          05 data1                                              PIC    X
      -    X.
      
       PROCEDURE DIVISION.
           DISPLAY STR
           GOBACK
           .
      
       END PROGRAM DVZZMFT0.
```

Here, the PIC clause starts at column 65 and we get those error messages :
- ID:27 Message:Syntax error : A group item cannot be empty. Severity:2 Line:7
- ID:27 Message:Syntax error : extraneous input 'DISPLAYSTR' expecting {ProgramIdentification, ProgramEnd, ClassIdentification, ClassEnd, FactoryEnd, ObjectIdentification, ObjectEnd, MethodEnd, ProcedureDivisionHeader, WorkingStorageSectionHeader, LocalStorageSectionHeader, LinkageSectionHeader, FileDescriptionEntry, DataDescriptionEntry, DataRedefinesEntry, DataRenamesEntry, DataConditionEntry, ExecStatement, FunctionDeclarationEnd, GlobalStorageSectionHeader} Severity:2 Line:12
- ID:27 Message:Syntax error : extraneous input 'PROCEDURE' expecting {separator, statement starting keyword, keyword} Severity:2 Line:11

**Second example**
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. DVZZMFT0.
      
       DATA DIVISION.
       WORKING-STORAGE SECTION.
      
       01 STR.
          05 data1                                                 PIC X
      -    X.
      
       PROCEDURE DIVISION.
           DISPLAY STR
           GOBACK
           .
      
       END PROGRAM DVZZMFT0.
```

With the PIC clause completey at the the end of the line, the error changes :
- ID:27 Message:Syntax error : extraneous input 'PROCEDURE' expecting {separator, statement starting keyword, keyword} Severity:2 Line:11

**Expected behavior**
There should be no diagnostics for these cases.

**Comment by smedilol** [comment]
@fm-117 Can you see if fix for #1777 resolve this issue ?

**Comment by fm-117** [comment]
> @fm-117 Can you see if fix for #1777 resolve this issue ?

It does not...
This is somewhat related to #1791 and #457, there is a problem in `Scanner.ScanTokensLineContinuationGroup` in the token attribution algorithm. In our case, the PeriodSeparator token is never added to the resulting lines and then ANTLR fails. This algorithm needs to be rewritten.
